src/index.js

```diff
@@ -60,68 +171,44 @@ document.addEventListener('DOMContentLoaded', () => {
     }
   });
 
-  renderTasks();
+  todoList.addEventListener('dragstart', (event) => {
+    event.dataTransfer.setData('text/plain', event.target.id);
+    event.target.classList.add('dragging');
+  });
 
-  const addTask = (taskText) => {
-    const task = {
-      text: taskText,
-      completed: false,
-    };
-    tasks.push(task);
-    const taskItem = createTaskItem(task);
-    todoList.appendChild(taskItem);
-  };
+  todoList.addEventListener('dragover', (event) => {
+    event.preventDefault();
+    const draggableTask = document.querySelector('.dragging');
+    const overTask = event.target.closest('.task-item');
+    if (draggableTask && overTask && draggableTask !== overTask) {
+      const fromIndex = Array.from(todoList.children).indexOf(draggableTask);
+      const toIndex = Array.from(todoList.children).indexOf(overTask);
+      reorderTask(fromIndex, toIndex);
+    }
+  });
 
-  const clearTasks = () => {
-    tasks = [];
-    todoList.innerHTML = '';
-  };
+  todoList.addEventListener('dragend', () => {
+    const draggableTask = document.querySelector('.dragging');
+    if (draggableTask) {
+      draggableTask.classList.remove('dragging');
+    }
+  });
+
+  const clearAllBtn = document.querySelector('.clear-all-btn');
+  clearAllBtn.addEventListener('click', removeCompletedTasks);
 
   const listForm = document.getElementById('list-form');
   const inputField = document.getElementById('input-field');
-  const clearAllBtn = document.querySelector('.clear-all-btn');
 
   listForm.addEventListener('submit', (event) => {
     event.preventDefault();
-    const taskText = inputField.value.trim();
-    if (taskText !== '') {
-      addTask(taskText);
+    const taskDescription = inputField.value.trim();
+    if (taskDescription !== '') {
+      addTask(taskDescription);
       inputField.value = '';
     }
   });
 
-  clearAllBtn.addEventListener('click', () => {
-    clearTasks();
-  });
-});
-
-const taskList = document.getElementById('todo-list');
-
-const moveTaskBelow = (taskElement) => {
-  const { nextElementSibling: nextTask } = taskElement;
-  if (nextTask) {
-    taskList.insertBefore(taskElement, nextTask.nextSibling);
-  }
-};
-
-const moveTaskAbove = (taskElement) => {
-  const { previousElementSibling: previousTask } = taskElement;
-  if (previousTask) {
-    taskList.insertBefore(taskElement, previousTask);
-  }
-};
-
-taskList.addEventListener('click', (event) => {
-  const { target } = event;
-  if (target.classList.contains('task-dots')) {
-    const taskElement = target.closest('.task-item');
-    if (taskElement) {
-      const { action } = target.dataset;
-      if (action === 'move-below') {
-        moveTaskBelow(taskElement);
-      } else if (action === 'move-above') {
-        moveTaskAbove(taskElement);
-      }
-    }
-  }
+  getTasksFromLocalStorage();
+  renderTasks();
 });
```

Comment:
You did an amazing job with the project so far. However, as per the project requirements, each task should also have a property `index` which should be set the value of the array length. And deleting a task should update all the remaining item's indexes, so they represent the current order and are unique. Kindly do so and then submit for a re-review.
![image](https://github.com/najibullahjafari/To-do-list/assets/33268927/082a8936-b080-478e-bd74-42f56eaefc86)